src/Bridge/Doctrine/Orm/Filter/DateFilter.php

```diff
@@ -185,7 +191,7 @@ protected function addWhere(QueryBuilder $queryBuilder, QueryNameGeneratorInterf
             ));
         }
 
-        $queryBuilder->setParameter($valueParameter, new \DateTime($value));
+        $queryBuilder->setParameter($valueParameter, $type->convertToDatabaseValue(new \DateTime($value), $queryBuilder->getEntityManager()->getConnection()->getDatabasePlatform()));
```

Comment:
What if `$type` is null?

Reply:
I had the condition before but removed it. Indeed `Type::getType` will throw if not found (in `getDateType`).

What I can do:

1. test for `hasType`
2. Use what we had before when `null`

Though, this method will only be called when the `Type` is available in the `DOCTRINE_DATE_TYPES` constant. Therefore, it's not really useful because it'll never be `null` ([those are all defined by default](https://github.com/doctrine/dbal/blob/master/lib/Doctrine/DBAL/Types/Type.php#L34)).